### DISCUSSION TITLE
订阅 Newsletter 后自动保存到简悦的稍后读

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3088] [服务]
> 将 Newsletter 的内容（邮件）自动导入到到简悦的稍后读。

# 描述

> 如果是 Newsletter 与简悦深度用户的话，请试试这个方法，可以实现类似 **Inoreader Pro 自带的邮箱收取 Newsletter** 的功能。

# 前置知识

> 如果想实现订阅 Newsletter 自动保存到稍后读，需要事先完成下面的配置。

1. [如何开通开放平台](https://github.com/Kenshin/simpread/discussions/2757)
2. [API 快照功能](https://github.com/Kenshin/simpread/discussions/2994)
3. [如何让稍后读支持读取本地缓存](https://github.com/Kenshin/simpread/discussions/2146)

# integromat

> 如果你用过 IFTTT / Zapier 的话，integromat 是一个比 IFTTT 更强大的自动化程序，更难能可贵的是，免费版的即可使用 Webhook 服务。（ _Zapier 的 Webhook 收费，虽然 IFTTT 不收费，但它的定制化无法跟 integromat 相比_） 

> 此教程需要你事先注册 https://www.integromat.com/

# 配置

> 以下是配置方案，如果用过类似的自动化服务，其实大多大同小异。

### 新建 

> 新建 **Create a new scenario** 并配置 **Mailhook 与 HTTP**

![image](https://user-images.githubusercontent.com/81074/142750553-ea6919b5-0c53-499d-909c-0d326183691f.png)

![image](https://user-images.githubusercontent.com/81074/142750564-c4d64100-c693-4e88-b280-84d19e76cb25.png)

![image](https://user-images.githubusercontent.com/81074/142750569-5958bd6d-cfab-420c-b88f-68fd23f358bf.png)

### 配置 HTTP

> 登录 [开放平台](http://simpread.pro/developer) 并得到你的 token 将下面代码的 `<你的 token>` 更换为你自己的 token

 ```
https://api-wrap.simpread.pro/api/service/webhook/<你的 token>?type=newsletter
```

![image](https://user-images.githubusercontent.com/81074/142750574-6f214593-62c1-4ece-b7c2-92c8d7c50577.png)

> 注意：下个截图的内容你可以直接用代码的代码粘贴过去。

> 将下图的代码复制到剪切板

```
{
    "title":"{{1.subject}}",
    "data":"{{encodeURL(1.html)}}"
}
```

> 再粘贴到下图所示位置

![image](https://user-images.githubusercontent.com/81074/142750705-4a43e014-710b-47ae-b0fd-af169b11f693.png)

> 上述设置无误后，根据下图所示测试

### 测试配置

> 发送邮件到 **上个步骤得到的 Email 地址**，如下图

> 这个邮箱地址大多是以 `xxxxxxx@hook.integromat.com` 结构的邮件地址

![image](https://user-images.githubusercontent.com/81074/142750869-a572273c-98ad-4dae-8148-cb7a5d5fda10.png)

> 如果上述配置无误的话，会出现下图的效果。（第三幅图）

![image](https://user-images.githubusercontent.com/81074/142752233-2c85bd19-d78f-4ef9-b6dd-55a83e351af5.png)

> 打开上图第三个幅图所示后出现类似下图的内容

![image](https://user-images.githubusercontent.com/81074/142752287-4d730df7-85f7-4d4f-927c-642745fca229.png)

> 并在你的本地 ouput 文件夹有一个叫做 `xxx-测试.html` 的文件。

![image](https://user-images.githubusercontent.com/81074/142752189-b00b2764-30d4-4df9-b18f-e7238e5f7519.png)

> 最后开启此 `scenario` 即可完成配置

![image](https://user-images.githubusercontent.com/81074/142752349-3c8ccec8-9cba-4aab-9183-8cca70960ed0.png)

# 如何使用

> 当配置成功后，你就得到一个 integromat 专门给你的 Email 地址，向此 Email 发邮件，邮件的内容就会出现在你的 `output` 文件夹。

> 如果是 Newsletter 的发送给你的话，自然正文就会出现在此，如何配置 Newsletter [请看这里](https://github.com/Kenshin/simpread/discussions/3088#discussioncomment-1676464)

# 关联

- [利用 Airtable 实现属于自己的链接公开分享集 #3003](https://github.com/Kenshin/simpread/discussions/3003)
- [Raindrop.io 加星后自动导入到简悦并生成本地快照 #3021](https://github.com/Kenshin/simpread/discussions/3021)
- [加入 Pocket 后自动保存到简悦的稍后读（包含快照） #3024](https://github.com/Kenshin/simpread/discussions/3024)
- [Instapaper 加星后自动保存到简悦的稍后读（包含快照） #3034](https://github.com/Kenshin/simpread/discussions/3034)
- [将简悦变成你的 RSS 阅读器（自动下载快照到本地） #3036](https://github.com/Kenshin/simpread/discussions/3036)
- [保存到 Pinboard.in 同时也保存到简悦（包括本地快照） #3048](https://github.com/Kenshin/simpread/discussions/3048)
- [Inoreader 加星后自动保存到简悦的稍后读以及生成本地快照 #3007](https://github.com/Kenshin/simpread/discussions/3007)
- [当加入稍后读后，自动发送到你的 Telegram Channel #3092](https://github.com/Kenshin/simpread/discussions/3092)
- [发送当前 URL 到飞书群（包括：加入稍后读或手动触发）#3104](https://github.com/Kenshin/simpread/discussions/3104)

# 引申

- [[教程\]如何开通开放平台 #2757](https://github.com/Kenshin/simpread/discussions/2757)
- [当在开放平台设置完毕后，如何测试配置是否成功？ #3020](https://github.com/Kenshin/simpread/discussions/3020)
- [授权登录（如何使用第三方登录简悦的高级账户） #1570](https://github.com/Kenshin/simpread/discussions/1570)
- [个人中心 #2959](https://github.com/Kenshin/simpread/discussions/2959)
- [免密码登录 #2974](https://github.com/Kenshin/simpread/discussions/2974)

**Comment by Kenshin** [comment]
# 如何配置 Newsletter

> 假设配置后的 Email 地址叫做 `abcde@hook.integromat.com` 目前有三种方式：


1. [通过邮箱转发](https://github.com/Kenshin/simpread/discussions/3088#discussioncomment-1676465)（首选）
2. [直接订阅 Newsletter](https://github.com/Kenshin/simpread/discussions/3088#discussioncomment-1676466)
3. [手动转发](https://github.com/Kenshin/simpread/discussions/3088#discussioncomment-1676468)

# 效果

> 下面的配置均以 订阅 [1000小食报](https://young.zhubai.love/) 为例，当成功订阅后，会将 Newsletter 保存到你的稍后读。


https://user-images.githubusercontent.com/81074/142753873-6fd5f533-675c-4d48-9138-04ec18666d12.mp4

**Comment by Kenshin** [comment]
# 通过邮箱转发（首选）

> 用 QQ 邮箱举例说明，并且已经订阅 [1000小食报](https://young.zhubai.love/) 为例

> 增加一个新的收信规则

![image](https://user-images.githubusercontent.com/81074/142752459-a2dfdb83-3dcc-4bcb-8c7f-e45934083d2d.png)

![image](https://user-images.githubusercontent.com/81074/142752548-1806d780-ea25-4672-a117-e7e1ea0b1c2b.png)

> 点击立即创建

![image](https://user-images.githubusercontent.com/81074/142752577-1e57bd85-3dc3-42e7-871b-0f13c0f897d1.png)

> 验证你提交的转发地址

> 如果上面步骤都没有错误的话，你应该会在 `output` 收到一个类似这样的文件

![image](https://user-images.githubusercontent.com/81074/142752595-7744ada8-23bd-414c-98d2-e9a458a8efd5.png)

> 用浏览器打开它

![image](https://user-images.githubusercontent.com/81074/142752613-900c1e49-0a4c-4ec6-b7b3-239d017cb555.png)

> 验证

![image](https://user-images.githubusercontent.com/81074/142752631-86304390-ed69-4fef-b58e-72cb519941f1.png)

> 回到收信规则，并刷新此页面

![image](https://user-images.githubusercontent.com/81074/142752667-a5f732b8-d639-4424-9e33-69d038818f1c.png)

**Comment by Kenshin** [comment]
# 直接订阅 Newsletter

> 直接订阅与转发类似， [1000小食报](https://young.zhubai.love/) 为例

![image](https://user-images.githubusercontent.com/81074/142752816-bf93999a-5839-45fc-9b04-6f5df68090e4.png)

> 如果配置无误的话，你应该会在 `output` 收到一个类似这样的文件

![image](https://user-images.githubusercontent.com/81074/142752832-e1b42deb-f34a-4853-ae0f-30132d69f7ca.png)

> 打开此文件并验证

![image](https://user-images.githubusercontent.com/81074/142752768-3ecba72d-bd3b-4a9c-a6d4-a1aaa6d2d255.png)

**Comment by Kenshin** [comment]
# 手动转发

> 收到转发就是将需要的内容转发到  `abcde@hook.integromat.com`  即可。

> 注意：为得到更好的视觉效果，在手动转发时，根据下图删除不必要的内容

![image](https://user-images.githubusercontent.com/81074/142752961-ca4d8da1-a5ce-48c1-8a24-296772916f10.png)

**Comment by yomizzz** [comment]
用QQ邮箱给那个webhook邮件地址发邮件后，在integromat页面点击run once，然后就一直卡在“the request was accepted，waiting for data.”，不知道为啥………
![图1](https://user-images.githubusercontent.com/22812174/145176043-c350f893-825b-490d-bc83-477f12b86bd2.png)
![图2](https://user-images.githubusercontent.com/22812174/145176270-28b9d61d-a624-417e-9a19-253626747f5a.png)

**Reply by Kenshin** [reply]
> the request was accepted，waiting for data.

这个提示意思是：它在等待接收数据。（用于测试你的配置是否正确）

你可以这样做：

> 随便找一封邮件，最好是你信箱中的 Newsletter，然后**转发到你操作时给你的邮箱中**。

![image](https://user-images.githubusercontent.com/81074/145189608-05afddcd-c138-4089-8dce-d0f9ee93c2ae.png)

**Reply by Kenshin** [reply]
> 我照你的建议做了，但就是会一直在等待接受数据，不会跳转到下一步。但是这个配置流程都是没问题的，难道是网络有问题？

我把你之前的那条删了，请在这里回复。 😁 

如果配置无误的话，应该没问题的，你通过我说的方式发送后，邮箱提示发送成功了吗？另外，只要发送成功了，下图对应的界面会有变化。

![image](https://user-images.githubusercontent.com/81074/145324524-b0ff76c0-5a25-4a71-b0e0-153f468ee93a.png)

**Reply by yomizzz** [reply]
邮件发送成功了，但还是会卡在那里。我刚又试了下，不跳转也没事，直接把测试停掉，然后开启这个 scenario，转发的邮件还是可以保存到简悦稍后读里。

**Reply by Kenshin** [reply]
emmm 挺奇怪的，不应该卡才对。不过，既然你能收到邮件就没问题了。

另外请注意：建议订阅一些低频或者晚上推送的 Newsletter，否则白天收到推送的话，如果恰好你也在加入稍后读，会有「冲突」的风险。

虽然，这种情况发生的很少，但还是提示下。

**Comment by tinainsz** [comment]
你好，转发成功但稍后读一直提示解析失败，是操作哪里有问题吗……

**Reply by Kenshin** [reply]
此功能需要配置 [知识库](https://www.yuque.com/kenshin/simpread/lglfy2) 你可以自查下自己的配置是否正确。

另外，如还是有问题的话，请提 https://github.com/kenshin/simpread/issues/new 不要在这里讨论。
